protocol/msgpack/base_benchmark_test.go

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package msgpack
+
+import (
+	"bytes"
+	"math/rand"
+	"testing"
+)
+
+var (
+	smallFloat64s  []float64
+	mediumFloat64s []float64
+	largeFloat64s  []float64
+)
+
+func BenchmarkEncodeFloat64NativeSmall(b *testing.B) {
```

Comment:
nit: take it or leave it, but could combine these into a single benchmark with a table, for example:

```
func BenchmarkFloat64(b *testing.B) {
  // initialize float slice

  benchmarks := []struct {
    name string
    vals []float64
    encoding encodingType
  } {
    {
      name: "EncodeNativeSmall"
      vals: smallFloat64a,
      encoding: nonPackedEncoding,
    },
    ...
  }

  for _, bm := range benchmarks {
    b.Run(bm.name, func(b *testing.B) {
      for i := 0; i < b.N; i++ {
        AppendFloat(dst[:0], bm.float, bm.fmt, bm.prec, bm.bitSize)
      }
    })
  }
}
```